### PR TITLE
Display stackex questions

### PR DESCRIPTION
**Pull request by werg**

### CODE REVIEW

**Comment by wanderingstan** [comment]
Getting error in `main.js` on first run: 

```
Uncaught TypeError: Cannot read property 'error' of undefined
    at errorId (vscode-resource:/Users/stan/code/ec/vscode-extension/media/main.js:8)
    at initErrorDiv (vscode-resource:/Users/stan/code/ec/vscode-extension/media/main.js:13)
    at window.addEventListener.event (vscode-resource:/Users/stan/code/ec/vscode-extension/media/main.js:54)
```

File: vscode-extension/src/extension.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ interface IFoundError {
   language: string; // Language error found in
   rawText: string; // Entire blob of error message
   title: string; // Best title to show
+  blobId?: number; // Id of the individual blob containing error
```
Comment (wanderingstan) [inline vscode-extension/src/extension.ts]: I don't think it's a good idea to assume a 1:1 relationship between an error and a blob of incoming text from tail. We're pretty sure there will be future cases where we're getting only one line at a time. Also related, the webview shouldn't "know" about blobs at all (as in `main.js:8`)—that should be abstracted into the concept of an "error". (Or "problem", or whatever we call these things. 😉)

File: vscode-extension/src/extension.ts [hunk 2]
```diff
@@ -221,6 +197,49 @@ class ErrorCentralPanel {
     });
   }
 
+  private _handleBlob(data: any, filePath: string) {
+    // to do make these unique across multiple sessions
```
Comment (wanderingstan) [inline vscode-extension/src/extension.ts]: Standard practise is to have "to do" always written `TODO`, as most IDE's will recognize this and highlight, plus it makes it easy to find them all with a search. See https://stackoverflow.com/questions/9586478/ide-comment-keywords/9586701

File: vscode-extension/src/extension.ts [hunk 3]
```diff
@@ -99,6 +101,9 @@ class ErrorCentralPanel {
 
     this._panel.webview.html = this._getHtmlForWebview();
     setInterval(() => this._checkForErrlogs(), 800);
+
+    // todo: when developing locally we need to not create an infinite loop by tailing
```
Comment (wanderingstan) [inline vscode-extension/src/extension.ts]: See 'todo' comment. At a minimum, code should be internally consistent.

**Comment by wanderingstan** [comment]
Nice! Does this require the local webserver to test or run?

File: vscode-extension/src/extension.ts (hunk 1 above)
Comment (werg) [reply vscode-extension/src/extension.ts]: I was thinking we would group various errors under one blob, if there were several.
This way, we could still display questions and answers and issues and other stuff, even if our regexes don't detect an error per se.
I.e. I think there are situations where we can't detect an identifiable error, but still could be helpful

**Comment by werg** [comment]
The extension should run ok even without the webserver, you just won't get results.
I'll try to find / replicate the error that you posted above.

File: vscode-extension/src/extension.ts (hunk 1 above)
Comment (wanderingstan) [reply vscode-extension/src/extension.ts]: Hmm, I'm thinking of the opposite case where one error spans multiple blobs. E.g. the error+stacktrace amount to 20 lines, which arrive as 20 separate blobs. (Right now that doesn't happen, but it's almost guaranteed to happen with some combination of language/compiler/ssh/remote/tailing/etc..)

In general, I think the concept of "blob" should be restricted to the case where we're trying to identify an error in a stream. Once we've identified an error, then create an error object (`IFoundError` at the moment) and from then on refer to it by its own id.

**Comment by wanderingstan** [comment]
Pushed a fix to get rendering working on `main.js`. Please test!

**Comment by wanderingstan** [comment]
What's the status of this PR? Looks like it was closed without being merged—I think there was good stuff in there? Less discuss on our call tomorrow.

**Pull request closed without merging**